### PR TITLE
dont allow scrolling when last line is visible

### PR DESCRIPTION
**Pull request by nnyyxxxx**
# watch the video

## Type of Change
- [x] New feature
- [x] Refactoring

for the color bleeding u see in the video and image its solved with this PR #723 

## Description
before:
image

after:

https://github.com/user-attachments/assets/49922891-1771-489e-ae8d-ed5dda91dc6d

### CODE REVIEW

File: tui/src/floating_text.rs [hunk 1]
```diff
@@ -185,7 +188,8 @@ impl FloatingText {
     }
 
     fn scroll_down(&mut self) {
-        if self.v_scroll + 1 < self.src.len() {
+        let visible_lines = self.get_visible_lines();
```
Comment (adamperkowski) [inline tui/src/floating_text.rs:191]:
```suggestion
        let visible_lines = self.frame_height.saturating_sub(2);
```

File: tui/src/floating_text.rs [hunk 2]
```diff
@@ -207,10 +211,16 @@ impl FloatingText {
             self.h_scroll += 1;
         }
     }
+
+    fn get_visible_lines(&self) -> usize {
+        self.frame_height.saturating_sub(2)
+    }
```
Comment (adamperkowski) [inline tui/src/floating_text.rs:217]:
```suggestion
```

**Review by adamperkowski — APPROVED** [review]
Works as intended. No bugs when resizing the window while in preview.

**Review by cartercanedy — APPROVED** [review]

**Comment by pace777** [comment]
needed one

**Review by ChrisTitusTech — APPROVED** [review]

**Pull request merged**